transform/flatten_mangler.go

```diff
@@ -112,11 +115,16 @@ func (f *FlattenMangler) flattenStruct(fieldPrefix, tagPrefix []string, sf refle
 		nestedK, _ := getUnderlyingKindType(nestedsf.Type)
 		switch nestedK {
 		case reflect.Struct:
-			flattened, err := f.flattenStruct(flattenedNames, flattenedTags, nestedsf)
-			if err != nil {
-				return out, err
+			// only flatten if it doesn't implement TextUnmarshaler
+			if !nestedsf.Type.Implements(textMValue) {
+				flattened, err := f.flattenStruct(flattenedNames, flattenedTags, nestedsf)
+				if err != nil {
+					return out, err
+				}
+				out = append(out, flattened...)
+				continue
 			}
-			out = append(out, flattened...)
+			fallthrough // don't flatten if it implements TextUnmarshaler
```

Comment:
I feel like we should either move the body of the `default` case outside the `switch` block if we want fall-through.

Added benefit: we can invert the `Implements` check so the body of the if-statement can have a body that consists entirely of `break` to break out of the switch-statement.

Reply:
Oh forgot about `break` working in switch statements. It's so much better with the break. Thank you!